src/patch_parser.py

```diff
@@ -156,6 +156,24 @@ def diff_best_match(self, string: List[str]) -> List[str]:
         return node.diff_best_match(string)
 
 
+@dataclasses.dataclass
+class HunkParserState:
+    _number_of_edits: int
+    _deleted_lines: int
+
+    def get_number_of_deleted_lines(self):
```

Comment:
nit: I'd drop the getters and setters here since we're using a dataclass.
We can have the callers interact with the fields directly, or use a helper where it makes sense (e.g. if we move gerrit_{orig,new}_line in here, then we could have a helper to increment them all)

Reply:
I see, to do everything in one go basically?

Reply:
re moving new fields in: yeah, I was thinking it'd be fine to do that in one go.
But if you'd prefer, we could do this separately.

One option could be to do this in two PRs, and another is to clean up the commits a bit so we merge this PR w/o squashing.
I.e. squash all the current commits, and then have a second commit that moves the remaining fields into `HunkParserState`